webapp/src/main/java/com/box/l10n/mojito/service/smartling/ThirdPartyTextUnitDTO.java

```diff
@@ -0,0 +1,100 @@
+package com.box.l10n.mojito.service.smartling;
+
+import com.box.l10n.mojito.entity.ThirdPartyTextUnit;
+
+public class ThirdPartyTextUnitDTO {
+    Long id;
+    String thirdPartyTextUnitId;
+    String mappingKey;
+    Long tmTextUnitId;
+    String repositoryName;
+    String assetPath;
+    String tmTextUnitName;
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getThirdPartyTextUnitId() {
+        return thirdPartyTextUnitId;
+    }
+
+    public void setThirdPartyTextUnitId(String thirdPartyTextUnitId) {
+        this.thirdPartyTextUnitId = thirdPartyTextUnitId;
+    }
+
+    public String getMappingKey() {
+        return mappingKey;
+    }
+
+    public void setMappingKey(String mappingKey) {
+        this.mappingKey = mappingKey;
+    }
+
+    public Long getTmTextUnitId() {
+        return tmTextUnitId;
+    }
+
+    public void setTmTextUnitId(Long tmTextUnitId) {
+        this.tmTextUnitId = tmTextUnitId;
+    }
+
+    public String getRepositoryName() {
+        return repositoryName;
+    }
+
+    public void setRepositoryName(String repositoryName) {
+        this.repositoryName = repositoryName;
+    }
+
+    public String getAssetPath() {
+        return assetPath;
+    }
+
+    public void setAssetPath(String assetPath) {
+        this.assetPath = assetPath;
+    }
+
+    public String getTmTextUnitName() {
+        return tmTextUnitName;
+    }
+
+    public void setTmTextUnitName(String tmTextUnitName) {
+        this.tmTextUnitName = tmTextUnitName;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof ThirdPartyTextUnitDTO)) return false;
```

Comment:
I think I mentioned that in another review. It is hard to really have meaningful "equals" implementation for those kind of entities: should it be based on the "id", the whole set of fields, etc. Since it is not obvious, we haven't been implementing them in the project. Instead we'd use an explicit comparator next to the code that needs it.

Reply:
This is for adding to a Set<ThirdPartyTextUnitDTO>. There is a meaningful equals here where each ThirdPartyTextUnit has unique hashcode, mapping key and text unit it matches to.